examples/example-base/example-base.cpp

```diff
@@ -170,6 +174,53 @@ int64_t getCurrentTime() { return std::chrono::high_resolution_clock::now().time
 
 int64_t getTimerFrequency() { return std::chrono::high_resolution_clock::period::den; }
 
+class DebugCallback : public IDebugCallback
+{
+public:
+    virtual SLANG_NO_THROW void SLANG_MCALL
+        handleMessage(DebugMessageType type, DebugMessageSource source, const char* message) override
+    {
+        const char* typeStr = "";
+        switch (type)
+        {
+        case DebugMessageType::Info:
+            typeStr = "INFO: ";
+            break;
+        case DebugMessageType::Warning:
+            typeStr = "WARNING: ";
+            break;
+        case DebugMessageType::Error:
+            typeStr = "ERROR: ";
+            break;
+        default:
+            break;
+        }
+        const char* sourceStr = "[GraphicsLayer]: ";
+        switch (source)
+        {
+        case DebugMessageSource::Slang:
+            sourceStr = "[Slang]: ";
+            break;
+        case DebugMessageSource::Driver:
+            sourceStr = "[Driver]: ";
+            break;
+        }
+        printf("%s%s%s\n", sourceStr, typeStr, message);
+#ifdef _WIN32
+        OutputDebugStringA(sourceStr);
+        OutputDebugStringA(typeStr);
+        OutputDebugStringW(String(message).toWString());
+        OutputDebugStringW(L"\n");
+#endif
+    }
+};
+
+void initDebugCallback()
+{
+    static DebugCallback callback = {};
+    gfxSetDebugCallback(&callback);
+}
```

Comment:
Yeah, it seems like the `IDevice::Desc` type could just have an `IDebugCallback* debugCallback = nullptr;` field, and then seting that field implicitly enables the debug layer, right? Optionally we could have a distinct `enum` field for setting the "level" of debugging that the user wants.